project-rome-docs/objectivec-api/remotesystems.commanding/MCDAppServiceConnection.md

```diff
@@ -0,0 +1,114 @@
+---
+title: MCDAppServiceConnection
+description:  This class manages a connection to a particular remote app service.
+keywords: microsoft, windows, iOS, iPhone, objectiveC, connected devices, Project Rome 
+---
+
+# class `MCDAppServiceConnection`
+
+```
+@interface MCDAppServiceConnection : NSObject
+```
+This class manages a connection to a particular remote app service.
+
+## Properties
+
+### appServiceInfo
+`@property(nonatomic, retain, nonnull) MCDAppServiceInfo* appServiceInfo;`
+
+Provides details about the target app service to connect to.
+
+### requestReceived 
+`@property(nonatomic, readonly, nonnull) MCDEvent<MCDAppServiceConnection*, MCDAppServiceRequestReceivedEventArgs*>* requestReceived;`
+
+Event for when a service request is received from a remote app.
+
+### serviceClosed 
+`@property(nonatomic, readonly, nonnull) MCDEvent<MCDAppServiceConnection*, MCDAppServiceClosedEventArgs*>* serviceClosed;`
+
+Event for when connection to the app service closes.
+
+**Note:** On the client app, the "app service closed" event is not raised when the `close` method is called (by either the
+client or the host); it is only raised on the client app for unexpected connection interruptions. On the host app, the event _is_
+raised when the client calls `close` (as well as for unexpected interruptions).
```

Comment:
I don't think this is quite right. When one side of the connection calls close, the other side will get serviceClosed if and only if the connection is proximal. If it's a cloud connection, serviceClosed will eventually fire, but not until like an hour later when things start to time out. But I don't think we should document that exact behavior.